### PR TITLE
remove all warnings from the unit tests

### PR DESCRIPTION
**Pull request by tardyp**
And make new warnings show as errors.

Over time, I got lazy at fixing warnings, this is a bad habbit, and now it is enforced.

This commit also include some fixes in the monkey patches, while I was looking at it and they are in the same files.

- enforce all warnings as errors (except docker import warning from worker/docker which happens if worker is in pip install -e mode)
   fixes the related warnings
- remove testcase_assert monkeypatch as we now only support python 2.7+
- add monkeypatch to reduce the default timeout to 5 seconds
- add monkeypatch to remove the buildbotNetUsage warning if we are in the unit tests

### CODE REVIEW

File: master/buildbot/test/__init__.py [hunk 1]
```diff
@@ -106,8 +107,13 @@
...
-# DeprecatedWorkerModuleWarning in tests as errors.
-# All DeprecatedWorkerNameWarning warnings should be explicitly catched too,
-# so fail on any DeprecatedWorkerAPIWarning.
-warnings.filterwarnings('error', category=DeprecatedWorkerAPIWarning)
+# All deprecated modules should be loaded, consider future warnings in tests as errors.
+# In order to not pollute the test outputs,
+# warnings in tests shall be forcefully tested with assertProducesWarning,
+# or shutdown using the warning module
+warnings.filterwarnings('error')
+# if buildbot_worker is installed in pi install -e mode, then the docker directory will
```
Comment (rutsky) [inline master/buildbot/test/__init__.py]: typo: "pip"

File: master/buildbot/test/__init__.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 
 import sys
 import warnings
+import exceptions
```
Comment (rodrigc) [inline master/buildbot/test/__init__.py]: import exceptions doesn't work on Python 3.

File: master/buildbot/test/__init__.py [hunk 3]
```diff
@@ -106,8 +107,16 @@
...
+# All deprecated modules should be loaded, consider future warnings in tests as errors.
+# In order to not pollute the test outputs,
+# warnings in tests shall be forcefully tested with assertProducesWarning,
+# or shutdown using the warning module
+warnings.filterwarnings('error')
+# if buildbot_worker is installed in pi install -e mode, then the docker directory will
+# match "import docker", and produce a warning.
+# We just suppress this warning instead of doing silly workaround.
+warnings.filterwarnings('ignore', "Not importing directory.*docker': missing __init__.py",
+                        category=exceptions.ImportWarning)
```
Comment (rodrigc) [inline master/buildbot/test/__init__.py]: I think `category=ImportWarning` will work.

**Comment by tardyp** [comment]
@rutsky I think I've fixed remaining stuff. Okay for merge?

**Comment by rutsky** [comment]
@tardyp approved.

**Comment by tardyp** [comment]
yeah! no more warnings!

**Pull request merged**